### PR TITLE
refactor(editor): Upgrade to jsPlumb 5

### PR DESCRIPTION
**Pull request by OlegIvaniv**

### CODE REVIEW

**Comment by netroy** [comment]
There seem to be a lot of unrelated changes in this branch. I'm assuming because of a bad merge?
Can you please remove those changes?

**Review by netroy — DISMISSED** [review]
There seem to be a lot of unrelated changes in this branch. I'm assuming because of a bad merge?
Can you please remove those changes?

File: packages/editor-ui/src/views/NodeView.vue [hunk 1]
```diff
@@ -765,6 +789,7 @@ export default mixins(
 				this.workflowsStore.setWorkflowPinData(data.workflowData.pinData);
 			}
 
+			// await new Promise((resolve) => setTimeout(resolve, 5000));
```
Comment (mutdmour) [inline packages/editor-ui/src/views/NodeView.vue:792]: to clean up

File: packages/editor-ui/src/views/NodeView.vue [hunk 2]
```diff
@@ -3296,11 +3316,13 @@ export default mixins(
 							});
 						}
 					}
+
+					const plumbNode = this.instance?.getEndpoint;
```
Comment (mutdmour) [inline packages/editor-ui/src/views/NodeView.vue:3320]: can be deleted?

File: packages/editor-ui/src/plugins/connectors/N8nCustomConnector.ts [hunk 3]
```diff
@@ -0,0 +1,586 @@
...
+				paintInfo.startStubX,
+				paintInfo.startStubY,
+				paintInfo.endStubX,
+				paintInfo.endStubY,
+			] as StubPositions;
+		}
+	},
+};
+
+export class N8nConnector extends AbstractConnector {
```
Comment (mutdmour) [inline packages/editor-ui/src/plugins/connectors/N8nCustomConnector.ts:142]: nice job here.. know this was not simple

**Review by mutdmour — COMMENTED** [review]
awesome job.. nitpicks are suggestions, feel free to address what you agree with.. also found a couple of bugs related to stickies, added to reviews page.

File: packages/editor-ui/src/views/NodeView.vue [hunk 4]
```diff
@@ -2261,25 +2201,103 @@ export default mixins(
...
+
+					Object.values(connection.overlays).forEach((overlay) => {
+						if (!overlay.canvas) return;
+						this.instance?.repaint(overlay.canvas);
+					});
+				});
+			});
+			this.instance.bind(EVENT_CONNECTION_MOUSEOVER, (connection: Connection) => {
+				try {
+					if (this.exitTimer !== undefined) {
```
Comment (mutdmour) [inline packages/editor-ui/src/views/NodeView.vue:2241]: the timers worked better when they were not global, but tied to connection state.. for example if you put two connections next to each other and hover quickly between them, it's easy to get stuck in a hover state without actions.. not the biggest issue..

File: packages/editor-ui/src/utils/nodeViewUtils.ts [hunk 5]
```diff
@@ -503,22 +519,26 @@ export const getBackgroundStyles = (
...
+export const showConnectionActions = (connection: Connection) => {
+	if (!connection) return;
+
+	showOverlay(connection, OVERLAY_CONNECTION_ACTIONS_ID);
+	hideOverlay(connection, OVERLAY_RUN_ITEMS_ID);
+	if (!getOverlay(connection, OVERLAY_RUN_ITEMS_ID)) {
+		hideOverlay(connection, OVERLAY_MIDPOINT_ARROW_ID);
 	}
+
+	connection.instance.repaintEverything();
```
[the remark this answers is not in the document]
Comment (OlegIvaniv) [reply packages/editor-ui/src/utils/nodeViewUtils.ts]: Good point, I've changed it to only repaint relevant connections

**Review by mutdmour — DISMISSED** [review]

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.214.0`

**Pull request merged**